src/Microsoft.AspNetCore.Mvc.Razor/Compilation/CompiledViewDescriptor.cs

```diff
@@ -1,13 +1,60 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
 using System.Collections.Generic;
+using Microsoft.AspNetCore.Mvc.Razor.Internal;
+using Microsoft.AspNetCore.Razor.Hosting;
 using Microsoft.Extensions.Primitives;
 
 namespace Microsoft.AspNetCore.Mvc.Razor.Compilation
 {
+    /// <summary>
+    /// Represents a compiled Razor View or Page.
+    /// </summary>
     public class CompiledViewDescriptor
```

Comment:
Unfortunately CVD is a public and appears in a bunch of places. I wanted to make `RazorCompiledItem` the new primary representation throughout MVC, but that's just not going to be feasible. 

We're also maintaining compatibility with MVC 2.0 here by continuing to process the old attributes. 

I'm considering changing it so that types are either new attributes OR old attributes. Right now the MVC codegen does both, and it's not ideal, but that would technically be a breaking change to take away 😢.

Reply:
What I mean to say is that I was hoping this code could be cleaned up quite a lot, but most of the changes I would want to make would be very subtle breaks, so I decided not to do them.

Reply:
This isn't the cleanest, but it will get us far enough to get the checksums.